### ISSUE TITLE
Traffic jam

### ISSUE THREAD

**Issue by Spikeone** [nicymike/serflings#15]
It's possible to jam your road system:
![image](https://user-images.githubusercontent.com/1276342/84404101-b6825f00-ac06-11ea-85c5-19c4ea581045.png)

**Comment by nicymike** [comment]
Yes, that is actually a bug. In the original settlers there are checks which send settlers to a different path if they are stuck like this.
I think this will be fixed after rewriting the logic for the settlers.

**Comment by Wakeboardtiger** [comment]
I also saw it happening in the original that one carriers at a crucial crossing point is getting dismissed after a while so the people can walk on. 
I figured following: In the case on the picture, the land digger with the orange hat have to walk right now in order to keep the flow going. Thats what would happen in the original. Because the Sherpa on top cannot change direction he has to keep on his walkway! And thats where in the original it can come to blocks as well, i think. 
Because if you would have a Sherpa on the lower flag as well now, at the position of the orange land digger, who serves the left walk way, then you created a block now, which cannot be resolved, since the Sherpas cannot change the walk ways. And then after a while one of the Sherpas getting dismissed from the walkway, running into the open countryside... 

Definition of Sherpa: a member of a Tibetan people living on the high southern slopes of the Himalayas in eastern Nepal and known for providing support for foreign trekkers and mountain climbers.
Or better known for carrying tourists up the Mount Everest and back.

**Comment by nicymike** [comment]
Thank you for the hints. I think the "turn around" thing is still missing in Serflings. I am using the DOS version of the game as source.

**Comment by Wakeboardtiger** [comment]
Btw I think due to this problem the Computer players get easily stuck, if this problem occurs to them, since they wont resolve the issue.

**Comment by nicymike** [comment]
I think you are right. As far as I know the source code, the computer does not care about traffic jams.

**Comment by nicymike** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Kein Erkennen von Siedlerstaus](https://trello.com/c/Uers7ZBp/1-kein-erkennen-von-siedlerstaus)

**Comment by nicymike** [comment]
This issue is fixed in the newest development version.
